### PR TITLE
fix: messages from non group members

### PR DESCRIPTION
**Pull request by MohamadJaara**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

after the changes to `MemberDetails`  model i had to change the logic in `GetMessagesForConversationUseCase` which reintroduced an old bug regarding fetching user info in a conversation

### Solutions

`ObserveUserListByIdUseCase` in `GetMessagesForConversationUseCase` and refactor the UI models to rely on `User` not Member

### Testing

WIP

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-761/2/) **failed**.

File: app/src/main/kotlin/com/wire/android/di/CoreLogicModule.kt [hunk 1]
```diff
@@ -192,6 +193,14 @@ class UseCaseModule {
     fun provideObserveConversationListDetailsUseCase(@KaliumCoreLogic coreLogic: CoreLogic, @CurrentAccount currentAccount: UserId) =
         coreLogic.getSessionScope(currentAccount).conversations.observeConversationListDetails
 
+    @ViewModelScoped
+    @Provides
+    fun observeObserveUserListByIdUseCaseProvide(
```
Comment (saleniuk) [inline app/src/main/kotlin/com/wire/android/di/CoreLogicModule.kt:198]: Shouldn't it be `provideObserveUserListByIdUseCase`? 😄

File: app/src/test/kotlin/com/wire/android/mapper/UIParticipantMapperTest.kt [hunk 2]
```diff
@@ -51,7 +51,8 @@ class UIParticipantMapperTest {
         memberDetails.userId == uiParticipant.id
                 && memberDetails.name == uiParticipant.name
                 && memberDetails.handle == uiParticipant.handle
-                && memberDetails.avatar(wireSessionImageLoader) == uiParticipant.avatarData
+                && TODO()
+                // && memberDetails.avatar(wireSessionImageLoader) == uiParticipant.avatarData
```
Comment (saleniuk) [inline app/src/test/kotlin/com/wire/android/mapper/UIParticipantMapperTest.kt:55]: Can't we just compare `memberDetails.user.avatar(wireSessionImageLoader) == uiParticipant.avatarData`? 🤔

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-761/4/) **failed**.

File: app/src/main/kotlin/com/wire/android/di/CoreLogicModule.kt (hunk 1 above)
Comment (MohamadJaara) [reply app/src/main/kotlin/com/wire/android/di/CoreLogicModule.kt:198]: will change it 😅

**Comment by AndroidBob** [comment]
Build [6](https://10.10.124.134/job/wire-android-reloaded/job/PR-761/6/) **failed**.

**Comment by AndroidBob** [comment]
Build [7](https://10.10.124.134/job/wire-android-reloaded/job/PR-761/7/) **failed**.

**Comment by AndroidBob** [comment]
Build [11](https://10.10.124.134/job/wire-android-reloaded/job/PR-761/11/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3866675)(3866675)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3866675)(3866675)-dev-release.apk)

**Review by saleniuk — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [15](https://10.10.124.134/job/wire-android-reloaded/job/PR-761/15/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3867420)(3867420)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3867420)(3867420)-dev-release.apk)

**Pull request merged**